common/src/main/java/com/microsoft/identity/common/internal/ui/webview/AzureActiveDirectoryWebViewClient.java

```diff
@@ -182,7 +189,15 @@ private boolean handleUrl(final WebView view, final String url) {
             } else {
                 Logger.info(methodTag,"This maybe a valid URI, but no special handling for this mentioned URI, hence deferring to WebView for loading.");
                 processInvalidUrl(url);
-                return false;
+
+                // Request headers submitted in the initial loadUrl will no longer be propagated for redirects on API31+.
+                // Override redirect and pass original request headers here.
+                if (mRequestHeaders != null && !mRequestHeaders.isEmpty()) {
+                    view.loadUrl(url, mRequestHeaders);
```

Comment:
do we need some additional check to verify if it is safe to send the headers to the redirected url?

Reply:
We should definitely add some checks given that these headers contain RT. We can create an allow list of domains (Microsoft Identity URLs) for which headers should be appended

Reply:
The allowed list can also be made such that it is only applicable to certain headers. For instance, the RT header is only appended if domains is part of allowed list, whereas the other headers can be appended regardless of the domain.

Reply:
On Windows, 302's are issued for Identity->Identity endpoints (e.g. Global ESTS -> MSA or UsGov ESTS -> Global ESTS).  Those are carrying all headers.

200's with a body of `document.location.replace` are being issued for ADFS cases, and headers are not being sent there.  I'm validating the behavior on Android now.

Reply:
Actually, swapping between using WAM and using in-proc network calls is changing the UsGov -> Global /authorize redirect between a 200 with WAM and a 302 in-proc.  So I don't think that's a good gate.  I was trying to avoid a whitelist, but maybe that's the way.  For speed of this change, maybe let's just gate to `login.live.com` for now to scope things down.